### PR TITLE
Correctness tests for whamm probes

### PR DESCRIPTION
**Pull request by ejrgilbert**
TODO:
- [x] write the tests
- [x] add test script
        - document issues as `failures.*` files
        - use the different tiers! (JIT/no-JIT)
- [x] add in matthew's tests from [other PR](https://github.com/titzer/wizard-engine/pull/223)

### CODE REVIEW

File: test/whamm/jit/correctness/req-data/localN.wasm
Comment (titzer) [inline test/whamm/jit/correctness/req-data/localN.wasm:1]: I think it's fine to just dump all the tests into one folder; no need to split them up, as the filenames will be enough of a clue what they do.

**Review by titzer — COMMENTED** [review]
Can you also add a shell script similar to the one in `test/wizeng/test.sh`? (In fact, you could probably copy that and start from it).

File: test/whamm/jit/correctness/req-data/localN.wasm
Comment (ejrgilbert) [reply test/whamm/jit/correctness/req-data/localN.wasm:1]: I'll do that after I finish writing them, I'm just doing the directory organization now to help my brain keep track of things.

Do you want me to add in a script even if they aren't passing at the moment? There's an issue with the `whamm` match logic it seems for some of these that I've written thus far (see the [README.md](https://github.com/titzer/wizard-engine/blob/8ccbb35606b368f9ef14d827ea0563b05b428d38/test/whamm/jit/correctness/README.md#issues)).

Comment (titzer) [reply test/whamm/jit/correctness/req-data/localN.wasm:1]: > Do you want me to add in a script even if they aren't passing at the moment?

Yes, there's a way to add expected failures, e.g. `test/regress.failures`.

**Pull request merged**